plansys2_core/include/plansys2_core/Action.hpp

```diff
@@ -0,0 +1,213 @@
+// Copyright 2025 Intelligent Robotics Lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PLANSYS2_CORE__ACTION_HPP_
+#define PLANSYS2_CORE__ACTION_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "plansys2_core/Types.hpp"
+#include "plansys2_msgs/msg/action.hpp"
+#include "plansys2_msgs/msg/durative_action.hpp"
+
+namespace plansys2
+{
+
+class Action : public plansys2_msgs::msg::Action
+{
+public:
+  Action()
+  : plansys2_msgs::msg::Action() {}
+  Action(const plansys2_msgs::msg::Action & action)  // NOLINT(runtime/explicit)
+  : plansys2_msgs::msg::Action(action)
+  {
+  }
+
+  bool operator==(const Action & action) const
+  {
+    if (this == &action) {return true;}
+    return parser::pddl::checkActionEquality(*this, action);
+  }
+};
+
+class DurativeAction : public plansys2_msgs::msg::DurativeAction
+{
+public:
+  DurativeAction()
+  : plansys2_msgs::msg::DurativeAction() {}
+  DurativeAction(const plansys2_msgs::msg::DurativeAction & action)  // NOLINT(runtime/explicit)
+  : plansys2_msgs::msg::DurativeAction(action)
+  {
+  }
+
+  bool operator==(const DurativeAction & action) const
+  {
+    if (this == &action) {return true;}
+    return parser::pddl::checkDurativeActionEquality(*this, action);
+  }
+};
+}  // namespace plansys2
+
+namespace std
+{
+template<>
+struct hash<plansys2::Action>
+{
+  std::size_t operator()(const plansys2::Action & action) const noexcept
+  {
+    std::size_t seed = 0;
+    hash_combine(seed, action.name);
+    hash_combine(seed, action.parameters.size());
+    hash_combine(seed, action.preconditions.nodes.size());
+    hash_combine(seed, action.effects.nodes.size());
+
+    for (const auto & param : action.parameters) {
+      hash_combine(seed, param.name);
+    }
+
+    for (auto node : action.preconditions.nodes) {
+      hash_combine(seed, node);
+    }
+    for (auto node : action.effects.nodes) {
+      hash_combine(seed, node);
+    }
+
+    return seed;
+  }
+};
+
+template<>
+struct hash<plansys2::DurativeAction>
+{
+  std::size_t operator()(const plansys2::DurativeAction & action) const noexcept
+  {
```

Comment:
Durative actions have a duration (that in PDDL might not be constant). How do you deal with this?